[DOCS-1690] Add docs for Restricted Viewer role

mdlinville opened the pull request:
[DOCS-1690] Add docs for Restricted Viewer role

- Add to Registry permissions table
- Add to Registry section of Teams documentation
- Add to aliases documentation
- Document key differences from Viewer role
- Document cross-registry permissions behavior
- Clarify ownership vs Registry roles
- Add SDK compatibility requirements (version 0.19.9+
- Update permissions table to use Unicode checkmark and X characters
- Improve SDK method descriptions to specify 'viewing artifact metadata only

Ready for peer review.

[DOCS-1690]: https://wandb.atlassian.net/browse/DOCS-1690?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

<!-- preview-links-comment -->
📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1565#issuecomment-3222102759)**

mdlinville commented on content/en/guides/core/registry/aliases.md:
```diff
@@ -31,7 +31,7 @@ Create one or more custom aliases for a specific artifact versions based on your
 - You might use a `best_model` alias to keep track of the best performing artifact model version.
 
-Any user with a [Member or Admin registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. If appropriate, use [protected aliases]({{< relref "aliases/#protected-aliases" >}}) to label and identify which artifact versions to protect from modification or deletion.
+Any user with a [Member or Admin registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. **Restricted Viewer** users cannot add or remove aliases. If appropriate, use [protected aliases]({{< relref "aliases/#protected-aliases" >}}) to label and identify which artifact versions  ...
```
```suggestion
Any user with a [Member or Admin registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the **Restricted Viewer** or **Viewer** roles cannot add or remove aliases.

[Protected aliases]({{< relref "aliases/#protected-aliases" >}}) provide a way to label and identify which artifact versions to protect from modification or deletion.
```
Convert the last sentence to a note

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -61,37 +61,48 @@ There are two different types of roles in W&B: [Team roles]({{< ref "/guides/mod
 {{% /alert %}}
 
+{{% alert title="Restricted Viewer role availability" %}}
+The **Restricted Viewer** role is currently available only in multi-tenant Cloud organizations. This role provides read-only access to registry content without the ability to create, update, or delete collections, automations, or other registry resources.
```
Multi-tenant Cloud

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -61,37 +61,48 @@ There are two different types of roles in W&B: [Team roles]({{< ref "/guides/mod
+{{% /alert %}}
+
 
 The proceeding table lists the different roles a user can have and their permissions:
```
```suggestion
The following table lists the different roles a user can have and permissions it confers:
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -61,37 +61,48 @@ There are two different types of roles in W&B: [Team roles]({{< ref "/guides/mod
-| Change registry visibility (Organization or Restricted)        | Admin            |        |        |   X   |
-| Add users to a registry                                        | Admin            |        |        |   X   |
-| Assign or change a user's role in a registry                   | Admin            |        |        |   X   |
+| Permission                                                     | Permission Group | Viewer | Restricted Viewer | Member | Admin | 
```
This table is confusing. Use a checkmark to indicate Yes, and an X to indicate No.

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
 | Team registry role | Individual registry role | Inherited registry role |
 | ------ | ------ | ------ | 
 | Viewer | Viewer | Viewer |
```
PTAL and make sure I've adjusted this correctly

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
 
+### Registry ownership
+
+Registry admins can assign a **Restricted Viewer** user as the registry's owner. Registry ownership is primarily for accountability purposes and does not confer any additional permissions beyond those granted by the user's assigned role. A Restricted Viewer who is also the registry owner will still have only read-only access to the registry content.
```
```suggestion
Any user can be designated as a registry's owner by a Registry admin, including a **Restricted Viewer** or a **Viewer**.  Registry ownership is primarily for accountability purposes and does not confer any additional permissions beyond those granted by the user's assigned role.
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+### SDK compatibility
+
+{{% alert title="SDK version requirement" %}}
+To use the Restricted Viewer role, you must use W&B SDK version 0.19.9 or higher. Older SDK versions will receive permission errors when trying to access artifacts in registries where users have Restricted Viewer roles.
```
```suggestion
To use the **Restricted Viewer** role, you must use W&B SDK version 0.19.9 or higher. Older SDK versions will receive permission errors when trying to access artifacts in registries where users have Restricted Viewer roles.
```
Render role names in bold

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+To use the Restricted Viewer role, you must use W&B SDK version 0.19.9 or higher. Older SDK versions will receive permission errors when trying to access artifacts in registries where users have Restricted Viewer roles.
+{{% /alert %}}
+
+The Restricted Viewer role affects SDK behavior in the following ways:
```
```suggestion
When a **Restricted Viewer** uses the SDK, certain functions are not available or work differently.
```

mdlinville replied:
Address all of this feedback systematically for the entries in this section. For example, adjust all of the entries that allow for viewing metadata only to align with the feedback in line 149.

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+
+The Restricted Viewer role affects SDK behavior in the following ways:
+
+- **`use_artifact()`**: Not available - will throw permission errors
```
```suggestion
- **`use_artifact()`**: Not available. Throw permission errors.
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+
+- **`use_artifact()`**: Not available - will throw permission errors
+- **`artifact.download()`**: Not available - will throw permission errors  
+- **`artifact.files()`**: Available but files cannot be downloaded
```
```suggestion
- **`artifact.files()`**: The artifact's directory structure and file names can be viewed, but files cannot be downloaded and their contents cannot be viewed.
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+- **`artifact.download()`**: Not available - will throw permission errors  
+- **`artifact.files()`**: Available but files cannot be downloaded
+- **`artifact.file()`**: Available but files cannot be downloaded
+- **`artifact.get()`**: Available for metadata access
```
```suggestion
- **`artifact.get()`**: Available for viewing artifact metadata only.
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+A user can have different roles in different registries. For example, a user can be a **Restricted Viewer** in Registry A but a **Viewer** in Registry B. In this case:
+
+- The same artifact linked to both registries will have different access levels
+- In Registry A: Cannot download files or use the artifact
```
```suggestion
- In Registry A, the user is a Restricted Viewer and cannot download files or use the artifact.
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+
+- The same artifact linked to both registries will have different access levels
+- In Registry A: Cannot download files or use the artifact
+- In Registry B: Can download files and use the artifact normally
```
```suggestion
- In Registry B, the user is a **Viewer** and can download files and use the artifact.
```

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,15 +118,49 @@ The proceeding table demonstrates the highest level of permission in the event o
+- The same artifact linked to both registries will have different access levels
+- In Registry A: Cannot download files or use the artifact
+- In Registry B: Can download files and use the artifact normally
+- Access is determined by the context in which the artifact is accessed (which registry collection)
```
```suggestion
In other words, access is determined by the registry in which the artifact is accessed.
```

mdlinville reviewed (commented):
Partial feedback

mdlinville committed:
[DOCS-1690] Add Restricted Viewer role documentation

    - Add to Registry permissions table
    - Add to Registry default roles table
    - Add to Registry inherited permissions table
    - Add to Registry section of Teams documentation
    - Add to aliases documentation
    - Document key differences from Viewer role
    - Document cross-registry permissions behavior
    - Clarify ownership vs Registry roles
    - Add SDK compatibility requirements (version 0.19.9+
    - Update permissions table to use Unicode checkmark and X characters
    - Improve SDK method descriptions to specify 'viewing artifact metadata only'
    - Info architecture adjustments

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+6. Click the **...** action menu at the end of the row, then click **Make owner**.
+
+
+## Configure Registry roles
```
Moved this up from the bottom so it can be near the other tasks and before the conceptual details.

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+6. In the **Registry role** column, click the user's role. 
+7. From the dropdown, select the role you want to assign to the user.
+
+## Details about Registry roles
```
Pushed this conceptual info downward

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
 {{% /alert %}}
 
-## Registry roles
+### Change the owner
```
Added the task details for changing the owner

ibindlish replied:
🙏

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+
+| Entity | Dedicated Cloud or Self-Managed<br />Default registry role | Multi-tenant Cloud<br />Default registry role |
+| ----- | ----- | ----- |
+| Team | Viewer | Restricted Viewer |
```
PTAL to make sure these details are correct for the SaaS rollout phase

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
 
-{{% alert title="W&B role types" %}}
-There are two different types of roles in W&B: [Team roles]({{< ref "/guides/models/app/settings-page/teams.md#team-role-and-permissions" >}}) and [Registry roles]({{< relref "configure_registry.md#configure-registry-roles" >}}).
+{{% alert title="Restricted Viewer role availability" %}}
```
I'll adjust this if we decide to hold the docs until it's self-serve for SaaS

ibindlish replied:
Sorry if this is a noob question - but what you mean by 'self-serve for SaaS'?

mdlinville replied:
Invite-only: We need to enable it for each org (I think this is the current state?) Self-serve: Anyone can turn on Restricted Viewer (self service), or everyone has access to Restricted Viewer (fully rolled out). The important thing is for the reader to know how to get access to the feature at the time that we publish these docs.

ibindlish replied:
Ah, understood - thank you. Yes, right now its invite-only. The docs will probably go out sooner than when the feature is fully rolled out on SaaS because we mean to do roll it out to ~30% of SaaS each week, over 3 weeks.

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+### Role permissions
+The following table lists each Registry role, along with the permissions provided by each role:
+
+| Permission                                                     | Permission Group | Restricted Viewer<br />(Multi-tenant Cloud, by invitation) | Viewer | Member | Admin | 
```
PTAL at this table, where I add a Restricted Viewer column before Viewer. I also swap out text X for Yes and blank for No, with unicode check character and blank for better legibility.

ibindlish replied:
lgtm, thanks!

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -117,11 +153,29 @@ For example, in the proceeding image Alex inherits **Member** role privileges be
-6. In the **Registry role** column, click the user's role. 
-7. From the dropdown, select the role you want to assign to the user.
\ No newline at end of file
+### SDK compatibility
```
PTAL and make sure these details are correct

ibindlish replied:
A couple of corrections:

`artifact.file()` is not available for a restricted viewer
`artifact.files()` is not available for a restricted viewer
I don't think `artifact.json_encode()` needs a special mention; IIRC, it was introduced for a launch-specific feature

mdlinville replied:
Done

ibindlish replied:
Thank you!

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -117,11 +153,29 @@ For example, in the proceeding image Alex inherits **Member** role privileges be
+- **`artifact.json_encode()`**: Available for viewing artifact metadata only.
+- **`artifact.verify()`**: Available for viewing artifact metadata only.
+
+### Cross-registry permissions
```
PTAL and make sure I explained this properly

ibindlish replied:
lgtm

mdlinville reviewed (commented):
Signposts for reviewers

ibindlish commented on content/en/guides/core/registry/aliases.md:
```diff
@@ -31,7 +31,13 @@ Create one or more custom aliases for a specific artifact versions based on your
+[Protected aliases]({{< relref "aliases/#protected-aliases" >}}) provide a way to label and identify which artifact versions to protect from modification or deletion.
+
+{{% alert %}}
+[Protected aliases]({{< relref "aliases/#protected-aliases" >}}) provide a way to label and identify which artifact versions to protect from modification or deletion.
```
**question (non-blocking):** Was the snippet about protected aliases intentionally added twice here?

mdlinville replied:
No, I was dithering about having it in an alert or not and didn't clean it up. Thanks.

ibindlish commented on content/en/guides/core/registry/aliases.md:
```diff
@@ -84,7 +90,7 @@ run.link_artifact(
 Use a [protected alias]({{< relref "aliases/#protected-aliases" >}}) to both label and identify artifact versions that should not be modified or deleted. For example, consider using a `production` protected alias to label and identify artifact versions that are in used in your organization's machine learning production pipeline.
 
-[Registry admins]({{< relref "/guides/core/registry/configure_registry/#registry-roles" >}}) and [service accounts]({{< relref "/support/kb-articles/service_account_useful" >}}) with the Admin role can create protected aliases and add or remove protected aliases from an artifact version. Members and Viewers cannot unlink a protected version or delete a collection that contains a protected . See [ ...
+[Registry admins]({{< relref "/guides/core/registry/configure_registry/#registry-roles" >}}) and [service accounts]({{< relref "/support/kb-articles/service_account_useful" >}}) with the **Admin** role can create protected aliases and add or remove protected aliases from an artifact version. **Members** and **Viewers** cannot unlink a protected version or delete a collection that contains a prote ...
```
**suggestion (non-blocking):** Should the `Restricted viewer` role be mentioned here too?

mdlinville replied:
Done

ibindlish commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+
+## Details about Registry roles
+
+There are two different types of roles in W&B: [Team roles]({{< ref "/guides/models/app/settings-page/teams.md#team-role-and-permissions" >}}) and [Registry roles]({{< relref "configure_registry.md#configure-registry-roles" >}}). Your role in a team has no impact or relationship to your role in any registry.
```
**suggestion:** There are also project roles, but it's probably best not to mention them here. Instead of specifying that there are 2 types of roles in W&B, I would just mention the last part of this block, that `one's role in a team has no impact/relationship to their role in any registry`.

mdlinville replied:
Done

ibindlish commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+| Entity | Dedicated Cloud or Self-Managed<br />Default registry role | Multi-tenant Cloud<br />Default registry role |
+| ----- | ----- | ----- |
+| Team | Viewer | Restricted Viewer |
+| Member (non admin) | Viewer | Restricted Viewer |
```
```suggestion
| User (non org-admin) | Viewer | Restricted Viewer |
```

Since `member` is also a role, i think using it here might be confusing.

mdlinville replied:
Done a little differently from the suggestion

ibindlish commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+- Cannot download artifact files or access file contents.
+- Cannot use artifacts with `use_artifact()` in the W&B SDK.
+- Cannot add or remove aliases or tags from artifacts.
+- Cannot create, edit, or delete collections, automations, or other registry resources.
```
1, 4, and 5 are also disallowed for the `Viewer` role. So i think it might be better to just highlight the main differences:

- Cannot download artifact files or access file contents.
- Cannot use artifacts with `use_artifact()` in the W&B SDK.

mdlinville replied:
Done

ibindlish commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -107,8 +140,11 @@ The proceeding table demonstrates the highest level of permission in the event o
 | Team registry role | Individual registry role | Inherited registry role |
 | ------ | ------ | ------ | 
 | Viewer | Viewer | Viewer |
```
Is this table meant to be exhaustive? Based on the existing docs, doesn't seem like that was the intention so lgtm! Else, there are a couple more combinations that are missing here.

mdlinville replied:
I'm not really clear on what this table is trying to show. I guess which takes precedence between your team and individual registry role. So I think it should be something like this (I am guessing at the third column values for the new ones. I am assuming that the highest permission between team and user wins, rather than the lowest. Is that correct?)

```text
| Restricted Viewer | Restricted Viewer | Restricted Viewer
| Restricted Viewer | Viewer                   | Viewer
| Restricted Viewer | Member                | Member
| Restricted Viewer | Admin                    | Admin
| Viewer                   | Restricted Viewer | Viewer
| Viewer                   | Viewer                    | Viewer
| Viewer                   | Member                 | Member
| Viewer                   | Admin                    | Admin
| Member                | Restricted Viewer | Member
| Member                | Viewer                   | Member
| Member                | Member                | Member
| Member                | Admin                    | Member
| Admin                   | Restricted Viewer | Admin
| Admin                   | Viewer                   | Admin
| Admin                   | Member                 | Admin
| Admin                   | Admin                    | Admin
```

Whether that is correct or not, there might be a more efficient way to show this. WDYT?

ibindlish replied:
Yup, you're completely right! And I agree, we could convey this more efficiently. Maybe the table isn't even required TBH.

ibindlish commented on content/en/guides/models/app/settings-page/teams.md:
```diff
@@ -76,14 +76,18 @@ Team settings allow you to manage the settings for your team and its members. Wi
-| Download models             | X         | X           | X              | X |
-|Add or remove Registry Admins  |           |             | X              | X | 
-|Add or remove Protected Aliases|           |             | X              |   | 
+| Permissions                | View-Only | Team Member | Restricted Viewer | Registry Admin | Team Admin | 
```
This `Registry` section refers to the legacy model registry and its access controls. Since registry roles aren’t relevant in team settings, it should be safe to remove this section entirely.

mdlinville replied:
Done

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -117,11 +153,29 @@ For example, in the proceeding image Alex inherits **Member** role privileges be
+- **`artifact.get_path()`**: Available for viewing artifact metadata only.
+- **`artifact.get_entry()`**: Available for viewing artifact metadata only.
+- **`artifact.json_encode()`**: Available for viewing artifact metadata only.
+- **`artifact.verify()`**: Available for viewing artifact metadata only.
```
Some of these are purposely hidden. See the Python SDK docs for which should and should not be exposed: https://docs.wandb.ai/ref/python/sdk/classes/artifact/

mdlinville replied:
After applying Ishita's feedback I verified and all of the remaining ones are referenced in the SDK docs. PTAL and let me know if this is the correct way to refer to these functions.

ngrayluna replied:
Thanks. Python Classes are capitalized, so each that start with "artifact" need to be like: `Artifact.download()`

Which `use_artifact()` is this referring to, btw? 
https://docs.wandb.ai/ref/python/sdk/classes/run/#method-runuse_artifact
https://docs.wandb.ai/ref/python/public-api/runs/#method-runuse_artifact

ibindlish replied:
This one: https://docs.wandb.ai/ref/python/sdk/classes/run/#method-runuse_artifact

mdlinville replied:
Fixed the formatting and turned each into a link so it is clear which method we mean.

mdlinville committed:
Feedback from Ishita and Noah

ibindlish commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,94 @@ A registry admin can remove individual users or entire teams from a registry. To
+| Entity                                 | Default registry role<br />(Dedicated Cloud / Self-Managed) | Default registry role<br />(Multi-tenant Cloud) |
+|----------------------------------------|-------------------------------------------------------------|-------------------------------------------------|
+| Team                                   | Viewer                                                      | Restricted Viewer                               |
+| User or service account (non admin)    | Viewer                                                      | Restricted Viewer                               |
```
Ooo thanks for adding Service accounts to this table. But they cannot be Viewers or Restricted viewers. So their default role is `Member`

mdlinville replied:
I added a row for service accounts with a footnote below the table to explain the situation.

ibindlish reviewed (approved):
LGTM! Thank you for the quick turnaround!

mdlinville committed:
Simplify inheritance section

- Show the UI first before explaining the algorithm
- Improve the explanation, remove the table
- Add the organization scope to the considered roles
- Improve the heading for better search

mdlinville committed:
Fix and link API methods, add service account row and footnote to effective role section

trane293 commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,62 +39,95 @@ A registry admin can remove individual users or entire teams from a registry. To
+### Role permissions
+The following table lists each Registry role, along with the permissions provided by each role:
+
+| Permission                                                     | Permission Group | Restricted Viewer<br />(Multi-tenant Cloud, by invitation) | Viewer | Member | Admin | 
```
nit: in a future PR, I'd remove the `(Multi-tenant Cloud, by invitation)` from here.

mdlinville replied:
It will be removed when it is no longer limited in these ways. Thanks for the reminder. :)

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
 {{% /alert %}}
 
-## Registry roles
+### Change the owner
```
The other H3 before this are about team management. Proposing adding "...of a registry" or something similar to distinguish between team management and changing the owner of a registry.

```suggestion
### Change the owner of a registry
```

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+
+
+### Effective (inherited) Registry role
+When viewing the registry's membership list, the W&B App shows each user's effective registry role next to the role dropdown in their row.
```
```suggestion
When viewing the registry's membership list, the W&B App shows each user's effective (inherited) registry role next to the role dropdown in their row.
```

mdlinville replied:
Your two suggestions on line 133 are a bit in conflict with each other and I can't figure out what end result you would prefer. Can you take a look?

mdlinville replied:
Merged into the next suggestion.

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+
+{{< img src="/images/registry/role_conflict.png" alt="Registry membership list showing the user's effective registry role" >}}
+
+A user's effective (inherited) role in a particular registry matches their _highest_ role among their role in the organization, the registry, and the team that owns the registry, whether inherited or explicitly assigned. For example:
```
```suggestion
A user's effective role in a particular registry matches their _highest_ role among their role in the organization, the registry, and the team that owns the registry, whether inherited or explicitly assigned. For example:
```

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+| Assign or change a user's role in a registry                   | Admin            |                   |        |        |   ✓   |
+
+
+### Effective (inherited) Registry role
```
Inherited is easier to understand. Consider just using that term and not "effective".

mdlinville replied:
I actually got confused thinking of it as inherited, which is why I changed it to "effective", but then I didn't want the reader to miss finding the info if they searched for "inherited" so I landed on having both. The way the UI is showing it and the way it ends up working out reads to me like "effective".

A quick differentiation I found is: "An inherited role is a role granted to a user or another role through a parent-child relationship within a role hierarchy, granting the recipient the privileges of the parent role. An effective role (or effective permissions) represents the cumulative set of all permissions a user holds, encompassing both their explicitly assigned roles and all roles inherited from groups, composite roles, or parent roles in a hierarchy"

I will take your advice though because you are more familiar with Registry users and how they think.

mdlinville replied:
Done

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+
+
+### Effective (inherited) Registry role
+When viewing the registry's membership list, the W&B App shows each user's effective registry role next to the role dropdown in their row.
```
Let's distinguish between inherited vs not inherited roles. 

```suggestion
The registry's membership list shows each user's inherited (effective) registry role (in light gray) next to the role dropdown in their row.
```

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+### SDK compatibility
+
+{{% alert title="SDK version requirement" %}}
+To use the W&B SDK to access artifacts as a **Restricted Viewer**, you must use W&B SDK version 0.19.9 or higher. Otherwise, SDK commands will result in permission errors.
```
```suggestion
To use the W&B SDK to access artifacts as a **Restricted Viewer**, you must use W&B SDK version 0.19.9 or higher. Otherwise, some SDK commands will result in permission errors.
```

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+When a **Restricted Viewer** uses the SDK, certain functions are not available or work differently.
 
-If there is a conflict, W&B displays the highest level of permissions next to the name of the user.
+The following functions are not available and result in permission errors:
```
What is listed below are methods, not functions

```suggestion
The following methods are not available and result in permission errors:
```

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+- [`Artifact.files()`]({{< relref "/ref/python/sdk/classes/artifact/#method-artifactfiles" >}})
 
-For example, in the proceeding image Alex inherits **Member** role privileges because they are a member of the `smle-reg-team-1` team.
+The following functions are limited to artifact metadata:
```
```suggestion
The following methods are limited to artifact metadata:
```

mdlinville committed:
Merge branch 'main' into DOCS-1690

mdlinville commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+| Assign or change a user's role in a registry                   | Admin            |                   |        |        |   ✓   |
+
+
+### Effective (inherited) Registry role
```
```suggestion
### Inherited Registry role
```

mdlinville committed:
Noah's feedback

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1690

mdlinville committed:
Apply suggestions from code review

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1690' into DOCS-1690

mdlinville committed:
Merge branch 'main' into DOCS-1690

mdlinville committed:
Merge branch 'main' into DOCS-1690

ngrayluna commented on content/en/guides/core/registry/aliases.md:
```diff
@@ -31,7 +31,12 @@ Create one or more custom aliases for a specific artifact versions based on your
 - You might use a `best_model` alias to keep track of the best performing artifact model version.
 
-Any user with a [Member or Admin registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. If appropriate, use [protected aliases]({{< relref "aliases/#protected-aliases" >}}) to label and identify which artifact versions to protect from modification or deletion.
+Any user with a [**Member** or **Admin** registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the **Restricted Viewer** or **Viewer** roles cannot add or remove aliases.
```
Suggestion: Also link Restricted and View to "guides/core/registry/configure_registry/#registry-roles"

mdlinville replied:
```suggestion
Any user with a [**Member** or **Admin** registry role]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) on a registry can add or remove a custom alias from a linked artifact in that registry. Users with the [**Restricted Viewer** or **Viewer** roles]({{< relref "guides/core/registry/configure_registry/#registry-roles" >}}) cannot add or remove aliases.
```

ngrayluna commented on content/en/guides/core/registry/aliases.md:
```diff
@@ -163,4 +168,4 @@ run.link_artifact(
 2. Next, you add two dataset PyTorch tensors to the artifact object with `wandb.Artifact.add_file()`. 
-3. Lastly, you link the artifact version to the `Model` collection in the `Zoo_Classifier_Models` registry with `link_artifact()`. You also add two custom aliases to the artifact version by passing  `production-us` and `production-eu` as arguments to the `aliases` parameter.
\ No newline at end of file
+3. Lastly, you link the artifact version to the `Model` collection in the `Zoo_Classifier_Models` registry with `link_artifact()`. You also add two custom aliases to the artifact version by passing  `production-us` and `production-eu` as arguments to the `aliases` parameter.
```
Why this highlighted in the PR review?

mdlinville replied:
Whitespace, A newline character was added at the end of the file. For some reason turning off whitespace in the diff viewer doesn't change how this line shows up. I asked Cursor locally because Copilot could not help me.

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
+### Change the owner of a registry
 
-Each user in a registry has a *registry role*, which determines what they can do in that registry. 
+A Registry admin can designate any member as a registry's owner, including a **Restricted Viewer** or a **Viewer**. Registry ownership is primarily for accountability purposes and does not confer any additional permissions beyond those granted by the user's assigned role.
```
```suggestion
A registry admin can designate any member as a registry's owner, including a **Restricted Viewer** or a **Viewer**. Registry ownership is primarily for accountability purposes and does not confer any additional permissions beyond those granted by the user's assigned role.
```

mdlinville replied:
Done, also one other instance in the file.

ngrayluna commented on content/en/guides/core/registry/configure_registry.md:
```diff
@@ -39,89 +39,132 @@ A registry admin can remove individual users or entire teams from a registry. To
 
-W&B automatically assigns a default registry role to a user or team when they are added to a registry. 
+To change the owner:
+1. Navigate to the Registry at https://wandb.ai/registry/.
```
Confirming specifying product URLs in the docs. (In the past you suggested I remove them).

mdlinville replied:
Done

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1690

mdlinville committed:
Update step to navigate to the Registry

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1690

mdlinville committed:
Fix another instance of 'registry admin'

mdlinville committed:
Update content/en/guides/core/registry/aliases.md

mdlinville committed:
Merge branch 'main' into DOCS-1690

ngrayluna reviewed (approved):
Thanks!

mdlinville committed:
Merge branch 'main' into DOCS-1690
